include/experimental/__p1673_bits/blas2_matrix_rank_1_update.hpp

```diff
@@ -48,6 +48,73 @@ namespace experimental {
 inline namespace __p1673_version_0 {
 namespace linalg {
 
+namespace {
+
+template <class Exec, class x_t, class y_t, class A_t, class = void>
+struct is_custom_matrix_rank_1_update_avail : std::false_type {};
+
+template <class Exec, class x_t, class y_t, class A_t>
+struct is_custom_matrix_rank_1_update_avail<
+  Exec, x_t, y_t, A_t,
+  std::void_t<
+    decltype(matrix_rank_1_update
+	     (std::declval<Exec>(),
+	      std::declval<x_t>(),
+	      std::declval<y_t>(),
+	      std::declval<A_t>()
+	      )
+	     )
+    >
+  >
+{
+  static constexpr bool value =
```

Comment:
I think I see what the idea is here -- the `void_t` thing checks that the expression is well formed, and the `value` thing back-ports to C++ < 20 (else you would use `same_as` to constrain, then inherit from `true_type`).

One potential issue is that not all specializations of the struct inherit from either `true_type` or `false_type`.  However, I don't think that's a big deal for an implementation detail, as long as developers know to check `value` and not try to check inheritance.